// @logto/js/lib/index.cjs
'use strict';

var fetchToken = require('./core/fetch-token.cjs');
var oidcConfig = require('./core/oidc-config.cjs');
var revoke = require('./core/revoke.cjs');
var signIn = require('./core/sign-in.cjs');
var signOut = require('./core/sign-out.cjs');
var userInfo = require('./core/user-info.cjs');
var callbackUri = require('./utils/callback-uri.cjs');
var errors = require('./utils/errors.cjs');
var idToken = require('./utils/id-token.cjs');
var accessToken = require('./utils/access-token.cjs');
var scopes = require('./utils/scopes.cjs');
var arbitraryObject = require('./utils/arbitrary-object.cjs');
var angular = require('./utils/angular.cjs');
var index = require('./consts/index.cjs');
var openid = require('./consts/openid.cjs');



exports.fetchTokenByAuthorizationCode = fetchToken.fetchTokenByAuthorizationCode;
exports.fetchTokenByRefreshToken = fetchToken.fetchTokenByRefreshToken;
exports.discoveryPath = oidcConfig.discoveryPath;
exports.fetchOidcConfig = oidcConfig.fetchOidcConfig;
exports.revoke = revoke.revoke;
exports.generateSignInUri = signIn.generateSignInUri;
exports.generateSignOutUri = signOut.generateSignOutUri;
exports.fetchUserInfo = userInfo.fetchUserInfo;
exports.parseUriParameters = callbackUri.parseUriParameters;
exports.verifyAndParseCodeFromCallbackUri = callbackUri.verifyAndParseCodeFromCallbackUri;
exports.LogtoError = errors.LogtoError;
exports.LogtoRequestError = errors.LogtoRequestError;
exports.OidcError = errors.OidcError;
exports.isLogtoRequestError = errors.isLogtoRequestError;
exports.isLogtoRequestErrorJson = errors.isLogtoRequestErrorJson;
exports.decodeIdToken = idToken.decodeIdToken;
exports.decodeAccessToken = accessToken.decodeAccessToken;
exports.withDefaultScopes = scopes.withDefaultScopes;
exports.withReservedScopes = scopes.withReservedScopes;
exports.isArbitraryObject = arbitraryObject.isArbitraryObject;
exports.buildAngularAuthConfig = angular.buildAngularAuthConfig;
exports.ContentType = index.ContentType;
Object.defineProperty(exports, "Prompt", {
	enumerable: true,
	get: function () { return index.Prompt; }
});
Object.defineProperty(exports, "QueryKey", {
	enumerable: true,
	get: function () { return index.QueryKey; }
});
Object.defineProperty(exports, "TokenGrantType", {
	enumerable: true,
	get: function () { return index.TokenGrantType; }
});
Object.defineProperty(exports, "ReservedResource", {
	enumerable: true,
	get: function () { return openid.ReservedResource; }
});
Object.defineProperty(exports, "ReservedScope", {
	enumerable: true,
	get: function () { return openid.ReservedScope; }
});
Object.defineProperty(exports, "UserScope", {
	enumerable: true,
	get: function () { return openid.UserScope; }
});
exports.buildOrganizationUrn = openid.buildOrganizationUrn;
exports.getOrganizationIdFromUrn = openid.getOrganizationIdFromUrn;
exports.idTokenClaims = openid.idTokenClaims;
exports.organizationUrnPrefix = openid.organizationUrnPrefix;
exports.userClaims = openid.userClaims;
exports.userinfoClaims = openid.userinfoClaims;
